test/generate-xml.sh

```diff
@@ -43,32 +43,40 @@ if [ ! -x $mbus_parse_hex ]; then
     exit 3
 fi
 
-for hexfile in "$directory"/*.hex;  do
-    if [ ! -f "$hexfile" ]; then
-        continue
-    fi
+generate_xml() {
+    directory="$1"
+    hexfile="$2"
+    mode="$3"
 
     filename=`basename $hexfile .hex`
```

Comment:
```suggestion
    filename=$(basename "$hexfile" .hex)
```

Shellcheck, a linter for shell scripts using $( ) instead of ` `, and wants you to add quotes where filenames possibly could contain spaces.

Reply:
Makes sense